src/riak_core_bucket.erl

```diff
@@ -192,14 +197,61 @@ bucket_nval_map(Ring) ->
 default_object_nval() ->
     riak_core_bucket:n_val(riak_core_bucket_props:defaults()).
 
+%% @private
+%% @doc For testing/debugging
+-spec all_n() -> [pos_integer(),...].
+all_n() ->
+    {ok, Ring} = riak_core_ring_manager:get_my_ring(),
+    all_n(Ring).
+
+-spec all_n(riak_core_ring()) -> [pos_integer(),...].
 all_n(Ring) ->
-    BucketNs = bucket_nval_map(Ring),
-    DefaultN = default_object_nval(),
-    AllN = lists:foldl(fun({_, N}, Acc) ->
-                               ordsets:add_element(N, Acc)
-                       end, [DefaultN], BucketNs),
+    BucketNVals = bucket_nvals(Ring),
+    BucketTypeNVals = bucket_type_nvals(),
+    ordsets:from_list(
```

Comment:
I do think to_list is, today at least, there to keep Dialyzer happy. However, it's certainly possible that someday Ericsson may change the internal representation, so I'd fix this to be to_list, which is essentially a no-op:

From https://github.com/erlang/otp/blob/maint/lib/stdlib/src/ordsets.erl#L70:

```erlang
to_list(S) -> S.
```